tests/storage/storage.py

```diff
@@ -23,40 +24,100 @@ def try_to_create_sr_with_missing_device(sr_type, label, host):
         return
     assert False, 'SR creation should not have succeeded!'
 
-def cold_migration_then_come_back(vm, prov_host, dest_host, dest_sr):
+def cold_migration_then_come_back(vm: VM, prov_host: Host, dest_host: Host, dest_sr: SR):
     """ Storage migration of a shutdown VM, then migrate it back. """
     prov_sr = vm.get_sr()
+    vdi_name = None
+
+    if not vm.is_windows:
+        # the vdi will be destroyed with the vm
+        vdi = prov_sr.create_vdi(virtual_size=1 * GiB)
+        vdi_name = vdi.name()
+        vm.connect_vdi(vdi, 'xvdb')
```

Comment:
(it looks like my previous comment did not get sent...)
Is it possible to use default "autodetect" instead of "xvdb" and use the return vdi-uuid to retrieve the name of the block device ?

Reply:
It might be possible, but I'm not sure how.
Do you fear `xvdb` could collide with another device?